emoji/src/EmojiSearchTable/EmojiSearchTable.js

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react';
+
+import style from './style.module.css';
+import EmojiSearch from '../EmojiSearch';
+import EmojiTable from '../EmojiTable';
+import emojiList from '../emojiList.json';
+
+export default function EmojiSearchTable(props) {
+    const [value, setValue] = useState('');
+
+    const saveValue = (event) => {
+        setValue(event.nativeEvent.target.value);
+    }
+
+    const filterEmoji = function (arr) {
```

Comment:
Use [useMemo](https://ru.reactjs.org/docs/hooks-reference.html#usememo) hook for calculated values
```
const filteredEmoji = useMemo(() => { 
        let result = emojiList.filter((item) => item.keywords.includes(value));
        if(result.length > 15) { 
            result = result.slice(0, 15)
        }
        return result;
    }, [emojiList])

   return (
   ....
      <EmojiTable list={filteredEmoji}/>
```